app/bitcoin/bitcoin.js

```diff
@@ -28,35 +42,60 @@ class BitcoinAddress {
     _hasBitcoinAddress() {
         return this.persistence.hasBitcoinAddress();
     }
+}
 
-    checkBalance() {
-        const address = this.persistence.getPublicKey();
-        return this.serverRequests.getBalance(address);
+
+class TransactionInput {
```

Comment:
Why not just have objects, rather than building classes? E.g., where you use `TransactionInput`, instead of `new TransactionInput(tx, index, value)`, do `{tx, index, value}`.

Rather than doing parameter validation by checking it's `variable.constructor`, like it is down in `BitcoinTransfer`, you can either check each value:

```
if ("tx" in input && "index" in input ...)
```

Or, the more standard JS way of doing things: let the nature of [Duck Typing](https://en.wikipedia.org/wiki/Duck_typing) handle things, and something will break if a parameter was missing. Besides, verifying `variable.constructor === TransactionInput` won't help anyway if someone initializes it with `undefined`.